pallets/dca/src/lib.rs

```diff
@@ -0,0 +1,1023 @@
+			Order::Buy { asset_in, .. } => asset_in,
+			Order::Sell { asset_in, .. } => asset_in,
+		};
+
+		T::Currency::reserve_named(
+			&NAMED_RESERVE_ID,
+			currency_for_reserve.into(),
+			who,
+			schedule.total_amount.into(),
+		)?;
+
+		Ok(())
+	}
+
+	fn unreserve_all_named_reserved_sold_currency(schedule_id: ScheduleId, who: &T::AccountId) -> DispatchResult {
+		let schedule = Schedules::<T>::get(schedule_id).ok_or(Error::<T>::ScheduleNotExist)?;
+		let sold_currency = Self::get_asset_in(&schedule.order);
+		T::Currency::unreserve_all_named(&NAMED_RESERVE_ID, sold_currency.into(), who);
+
+		Ok(())
+	}
+
+	fn get_asset_in(order: &Order<T::Asset>) -> <T as Config>::Asset {
+		let sold_currency = match order {
+			Order::Sell { asset_in, .. } => asset_in,
+			Order::Buy { asset_in, .. } => asset_in,
+		};
+		*sold_currency
+	}
+
+	fn get_asset_out(order: &Order<T::Asset>) -> <T as Config>::Asset {
+		let asset_out = match order {
+			Order::Sell { asset_out, .. } => asset_out,
+			Order::Buy { asset_out, .. } => asset_out,
+		};
+		*asset_out
+	}
+
+	fn weight_to_fee(weight: Weight) -> Balance {
+		// cap the weight to the maximum defined in runtime, otherwise it will be the
+		// `Bounded` maximum of its data type, which is not desired.
+		let capped_weight: Weight = weight.min(T::BlockWeights::get().max_block);
+		<T as pallet::Config>::WeightToFee::weight_to_fee(&capped_weight)
+	}
+
+	fn add_schedule_id_to_existing_ids_per_block(
+		next_schedule_id: ScheduleId,
+		blocknumber_for_schedule: <T as frame_system::Config>::BlockNumber,
+	) -> DispatchResult {
+		let schedule_ids =
+			ScheduleIdsPerBlock::<T>::get(blocknumber_for_schedule).ok_or(Error::<T>::NoScheduleIdsPlannedInBlock)?;
+		if schedule_ids.len() == T::MaxSchedulePerBlock::get() as usize {
+			let mut consequent_block = blocknumber_for_schedule;
+			consequent_block.saturating_inc();
+			Self::plan_schedule_for_block(consequent_block, next_schedule_id)?;
+			return Ok(());
+		} else {
+			Self::add_schedule_id_to_block(next_schedule_id, blocknumber_for_schedule)?;
+		}
+
+		Ok(())
+	}
+
+	fn add_schedule_id_to_block(
+		next_schedule_id: ScheduleId,
+		blocknumber_for_schedule: T::BlockNumber,
+	) -> DispatchResult {
+		ScheduleIdsPerBlock::<T>::try_mutate_exists(blocknumber_for_schedule, |schedule_ids| -> DispatchResult {
+			let schedule_ids = schedule_ids.as_mut().ok_or(Error::<T>::NoScheduleIdsPlannedInBlock)?;
+
+			schedule_ids
+				.try_push(next_schedule_id)
+				.map_err(|_| Error::<T>::InvalidState)?;
+			Ok(())
+		})?;
+
+		Ok(())
+	}
+
+	fn plan_schedule_for_block(b: T::BlockNumber, schedule_id: ScheduleId) -> DispatchResult {
+		if !ScheduleIdsPerBlock::<T>::contains_key(b) {
+			let vec_with_first_schedule_id = Self::create_bounded_vec(schedule_id)?;
+			ScheduleIdsPerBlock::<T>::insert(b, vec_with_first_schedule_id);
+		} else {
+			Self::add_schedule_id_to_existing_ids_per_block(schedule_id, b)?;
+		}
+		Ok(())
```

Comment:
you should be able to just use `try_append` for the schedules here (definitely if you change the storage to `ValueQuery`)
(your spill over would happen on failed try_append I imagine)